### PR TITLE
[OSDOCS-4774] ShiftStack 4.12 release notes

### PR DESCRIPTION
**Pull request by maxwelldb**
Version(s): 4.12

Issue: https://issues.redhat.com/browse/OSDOCS-4774

Link to docs preview: https://54280--docspreview.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-12-release-notes.html

QE review:
- [x] QE has approved this change.

Additional information: This PR is meant to represent contain all 4.12 ShiftStack release note updates that _are not_ part of https://github.com/openshift/openshift-docs/pull/45518 or https://github.com/openshift/openshift-docs/pull/53623

### CODE REVIEW

**Comment by maxwelldb** [comment]
Take a look, @MaysaMacedo @stephenfin @mandre @dulek @EmilienM @EricArrakis

File: release_notes/ocp-4-12-release-notes.adoc [hunk 1]
```diff
@@ -2139,6 +2158,11 @@ $ ./openshift-install wait-for install-complete
 
 * When using the Technology Preview OCI feature with the oc-mirror CLI plug-in, the mirrored catalog embeds all of the Operator bundles, instead of filtering only on those specified in the image set configuration file. (link:https://issues.redhat.com/browse/OCPBUGS-5085[*OCPBUGS-5085*])
 
+* Fully populated load balancers on {rh-openstack} can be stuck in a pending creation status. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. link:https://bugzilla.redhat.com/show_bug.cgi?id=2042976[*BZ#2042976*]
+
+* Bulk member updates on {rh-openstack} can return a 500 code in response to PUT requests. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. link:https://bugzilla.redhat.com/show_bug.cgi?id=2100135[*BZ#2100135*]
```
Comment (MaysaMacedo) [inline release_notes/ocp-4-12-release-notes.adoc:2163]:
```suggestion
* Bulk load-balancer member updates on {rh-openstack} can return a 500 code in response to PUT requests. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. link:https://bugzilla.redhat.com/show_bug.cgi?id=2100135[*BZ#2100135*]
```

File: release_notes/ocp-4-12-release-notes.adoc [hunk 2]
```diff
@@ -2139,6 +2158,11 @@ $ ./openshift-install wait-for install-complete
 
 * When using the Technology Preview OCI feature with the oc-mirror CLI plug-in, the mirrored catalog embeds all of the Operator bundles, instead of filtering only on those specified in the image set configuration file. (link:https://issues.redhat.com/browse/OCPBUGS-5085[*OCPBUGS-5085*])
 
+* Fully populated load balancers on {rh-openstack} can be stuck in a pending creation status. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. link:https://bugzilla.redhat.com/show_bug.cgi?id=2042976[*BZ#2042976*]
```
Comment (MaysaMacedo) [inline release_notes/ocp-4-12-release-notes.adoc:2161]:
```suggestion
* Fully populated load balancers on {rh-openstack} created with OVN driver can contain pool stuck in a pending creation status. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. link:https://bugzilla.redhat.com/show_bug.cgi?id=2042976[*BZ#2042976*]
```

**Comment by MaysaMacedo** [comment]
@dulek should the following bug also be highlighted in the RN https://bugzilla.redhat.com/show_bug.cgi?id=2143732?

**Comment by maxwelldb** [comment]
Also tagging @eurijon @rlobillo @itzikb-redhat

File: release_notes/ocp-4-12-release-notes.adoc [hunk 3]
```diff
@@ -2139,6 +2158,11 @@ $ ./openshift-install wait-for install-complete
...
+
+* Bulk load-balancer member updates on {rh-openstack} can return a 500 code in response to PUT requests. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. link:https://bugzilla.redhat.com/show_bug.cgi?id=2100135[*BZ#2100135*]
+
+
```
Comment (mandre) [inline release_notes/ocp-4-12-release-notes.adoc:2165]: There's [a known issue](https://issues.redhat.com/browse/OCPBUGS-5036) where the external cloud provider doesn't pick up the updated credentials after a rotation. A workaround exists that consists in killing the pod for them to consume the new credential. It could be problematic for customers on OpenStack where the external cloud provider has replaced the in-tree one.

Comment (maxwelldb) [reply release_notes/ocp-4-12-release-notes.adoc:2165]: Is this specific to ShiftStack, or does it impact other platforms, too?

Comment (mandre) [reply release_notes/ocp-4-12-release-notes.adoc:2165]: All platforms using external cloud provider. I can't find the list of what are these platforms in 4.12.

Comment (MaysaMacedo) [reply release_notes/ocp-4-12-release-notes.adoc:2165]: It looks like this is the [list](https://github.com/openshift/cluster-cloud-controller-manager-operator/blob/release-4.12/vendor/github.com/openshift/library-go/pkg/cloudprovider/external.go#L35-L40) of platforms using external cloud provider in 4.12 by default.

**Comment by EricArrakis** [comment]
I reviewed, although I don't have any useful comments :)

File: release_notes/ocp-4-12-release-notes.adoc [hunk 4]
```diff
@@ -647,6 +654,18 @@ As an administrator, you can now install the Network Observability Operator to o
...
+==== IPv6 for secondary network interfaces on {rh-openstack}
+
+IPv6 for secondary network intefaces is now supported in clusters that run on {rh-openstack}. 
+
+// TODO: For more information,see ${DOC_WHEN_MERGED} 
+
+[id="ocp-4-12-nw-shiftstack-udp-load-balancing"]
+==== UDP support for load balancers on {rh-openstack}
+
+UDP is now supported for load balancers for clusters that run on {rh-openstack}.
```
Comment (dulek) [inline release_notes/ocp-4-12-release-notes.adoc:667]: I'd rephrase it into "UDP protocol". But anyway in general we should say that OpenStack is now using the external cloud provider (cloud-controller-manager/CCM). This will tell way more for people actually knowing about Kubernetes. UDP LBs support is just one of the features we get by using CCM. Another one is e.g. ability to use OVN Octavia provider.

File: release_notes/ocp-4-12-release-notes.adoc [hunk 5]
```diff
@@ -2139,6 +2158,11 @@ $ ./openshift-install wait-for install-complete
 
 * When using the Technology Preview OCI feature with the oc-mirror CLI plug-in, the mirrored catalog embeds all of the Operator bundles, instead of filtering only on those specified in the image set configuration file. (link:https://issues.redhat.com/browse/OCPBUGS-5085[*OCPBUGS-5085*])
 
+* Fully populated load balancers on {rh-openstack} that are created with the OVN driver can contain pools that are stuck in a pending creation status. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. link:https://bugzilla.redhat.com/show_bug.cgi?id=2042976[*BZ#2042976*]
+
+* Bulk load-balancer member updates on {rh-openstack} can return a 500 code in response to PUT requests. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. link:https://bugzilla.redhat.com/show_bug.cgi?id=2100135[*BZ#2100135*]
```
Comment (dulek) [inline release_notes/ocp-4-12-release-notes.adoc:2163]: This only affects OSP17, should we say that?

**Review by dulek — CHANGES_REQUESTED** [review]
Also it makes sense to include information about BZ https://bugzilla.redhat.com/show_bug.cgi?id=2143732 as @MaysaMacedo suggests.

File: release_notes/ocp-4-12-release-notes.adoc (hunk 5 above)
Comment (maxwelldb) [reply release_notes/ocp-4-12-release-notes.adoc:2163]: @dulek Both, or just L2163? 

```suggestion
* Fully populated load balancers on {rh-openstack} 17 that are created with the OVN driver can contain pools that are stuck in a pending creation status. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. link:https://bugzilla.redhat.com/show_bug.cgi?id=2042976[*BZ#2042976*]

* Bulk load-balancer member updates on {rh-openstack} 17 can return a 500 code in response to PUT requests. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. link:https://bugzilla.redhat.com/show_bug.cgi?id=2100135[*BZ#2100135*]
```

File: release_notes/ocp-4-12-release-notes.adoc (hunk 4 above)
Comment (maxwelldb) [reply release_notes/ocp-4-12-release-notes.adoc:667]:
```suggestion
Resulting from the switch to an external cloud provider, UDP is now supported for load balancers for clusters that run on {rh-openstack}.
```

@dulek Am I wrong, or is the "P" in "UDP" for "Protocol?"

File: release_notes/ocp-4-12-release-notes.adoc (hunk 5 above)
Comment (dulek) [reply release_notes/ocp-4-12-release-notes.adoc:2163]: Aww, no, I'm wrong, it's all good as is. Sorry, so many bugs I got confused, this comment thread can be disregarded.

File: release_notes/ocp-4-12-release-notes.adoc (hunk 3 above)
Comment (maxwelldb) [reply release_notes/ocp-4-12-release-notes.adoc:2291]: See https://github.com/openshift/openshift-docs/pull/54280/commits/7d5e3a7cfcc39dc8a17d6d8ca4d39b1374173fa9

**Comment by maxwelldb** [comment]
Add health monitors KI: https://github.com/openshift/openshift-docs/pull/54280/commits/69a45399e51ff25843c58679c9e07000e4114369

File: release_notes/ocp-4-12-release-notes.adoc (hunk 3 above)
Comment (mandre) [reply release_notes/ocp-4-12-release-notes.adoc:2293]: Aren't we missing OpenStack in the list of affected platforms?

Comment (maxwelldb) [reply release_notes/ocp-4-12-release-notes.adoc:2293]: Added. :)

File: release_notes/ocp-4-12-release-notes.adoc [hunk 6]
```diff
@@ -2248,6 +2267,31 @@ $ ./openshift-install wait-for install-complete
...
+** {VirtProductName}
+** {rh-openstack}
+--
++
+As a workaround, restart openshift-cloud-controller-manager pods by running the following command:
++
+[source,terminal]
+----
+$ oc delete pods --all -n openshift-cloud-controller-manager
+----
```
Comment (maxwelldb) [inline release_notes/ocp-4-12-release-notes.adoc:2289]: FYI @imatza-rh

Comment (imatza-rh) [reply release_notes/ocp-4-12-release-notes.adoc:2289]: The W/A looks good. Thanks

Comment (maxwelldb) [reply release_notes/ocp-4-12-release-notes.adoc:2289]: TY!

File: release_notes/ocp-4-12-release-notes.adoc [hunk 7]
```diff
@@ -1267,6 +1332,11 @@ In {product-title} 4.12, support for {op-system} functionality is deprecated for
 
 While these hardware models remain fully supported in {product-title} 4.12, Red Hat recommends that you use later hardware models.
 
+[id="ocp-4-12-ne-kuryr"]
+==== Kuryr support for clusters that run on {rh-openstack}
+
+In {product-title} 4.12, support for Kuryr on clusters that run on {rh-openstack} is deprecated. In an upcoming release, support will be removed.
```
Comment (dulek) [inline release_notes/ocp-4-12-release-notes.adoc]: "In an upcoming release" suggests it can be 4.13 and we aim for removal to happen no earlier than 4.14.

Comment (maxwelldb) [reply release_notes/ocp-4-12-release-notes.adoc]: Should be addressed by https://github.com/openshift/openshift-docs/pull/54280/commits/0000456c104ec8b89042a487f75370979fa4a506

Comment (dulek) [reply release_notes/ocp-4-12-release-notes.adoc]: Yup, it is.

**Review by dulek — APPROVED** [review]
Looks okay from my side, we probably still need rest of the team to give their acks.

File: release_notes/ocp-4-12-release-notes.adoc [hunk 8]
```diff
@@ -2288,6 +2327,31 @@ $ ./openshift-install wait-for install-complete
 
 * When using the Technology Preview OCI feature with the oc-mirror CLI plug-in, the mirrored catalog embeds all of the Operator bundles, instead of filtering only on those specified in the image set configuration file. (link:https://issues.redhat.com/browse/OCPBUGS-5085[*OCPBUGS-5085*])
 
+* Fully populated load balancers on {rh-openstack} that are created with the OVN driver can contain pools that are stuck in a pending creation status. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2042976[*BZ#2042976*])
+
+* Bulk load-balancer member updates on {rh-openstack} can return a 500 code in response to PUT requests. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2100135[*BZ#2100135*])
+
+* Clusters that use external cloud providers can fail to retrieve updated credentials after rotation. The following platforms are affected:
```
Comment (MaysaMacedo) [inline release_notes/ocp-4-12-release-notes.adoc:2334]: @JoelSpeed @mandre Can you confirm if the listed platforms are affected by [this bug](https://issues.redhat.com/browse/OCPBUGS-5036)?

**Review by MaysaMacedo — COMMENTED** [review]
LGTM.

File: release_notes/ocp-4-12-release-notes.adoc (hunk 8 above)
Comment (JoelSpeed) [reply release_notes/ocp-4-12-release-notes.adoc:2334]: To my knowledge, this list is correct for 4.12 yes

File: release_notes/ocp-4-12-release-notes.adoc [hunk 9]
```diff
@@ -2288,6 +2327,31 @@ $ ./openshift-install wait-for install-complete
 
 * When using the Technology Preview OCI feature with the oc-mirror CLI plug-in, the mirrored catalog embeds all of the Operator bundles, instead of filtering only on those specified in the image set configuration file. (link:https://issues.redhat.com/browse/OCPBUGS-5085[*OCPBUGS-5085*])
 
+* Fully populated load balancers on {rh-openstack} that are created with the OVN driver can contain pools that are stuck in a pending creation status. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2042976[*BZ#2042976*])
+
+* Bulk load-balancer member updates on {rh-openstack} can return a 500 code in response to PUT requests. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2100135[*BZ#2100135*])
```
Comment (adellape) [inline release_notes/ocp-4-12-release-notes.adoc:2332]: Maybe markup `PUT`

File: release_notes/ocp-4-12-release-notes.adoc [hunk 10]
```diff
@@ -2288,6 +2327,31 @@ $ ./openshift-install wait-for install-complete
...
++
+--
+** {alibaba}
+** IBM Cloud VPC
+** IBM Power
+** {VirtProductName}
+** {rh-openstack}
+--
++
+As a workaround, restart openshift-cloud-controller-manager pods by running the following command:
```
Comment (adellape) [inline release_notes/ocp-4-12-release-notes.adoc:2344]: Markup `openshift-cloud-controller-manager`

File: release_notes/ocp-4-12-release-notes.adoc [hunk 11]
```diff
@@ -2288,6 +2327,31 @@ $ ./openshift-install wait-for install-complete
...
++
+[source,terminal]
+----
+$ oc delete pods --all -n openshift-cloud-controller-manager
+----
++
+(link:https://issues.redhat.com/browse/OCPBUGS-5036[*OCPBUGS-5036*])
+
+* There is a known issue when cloud-provider-openstack tries to create health monitors on OVN load balancers by using the API to create fully populated load balancers. These health monitors become stuck in a `PENDING_CREATE` status. After their deletion, associated load balancers are are stuck in a `PENDING_UPDATE` status. There is no workaround. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2143732[*BZ#2143732*])
```
Comment (adellape) [inline release_notes/ocp-4-12-release-notes.adoc:2353]: Should `cloud-provider-openstack` be marked up? 🤷

File: release_notes/ocp-4-12-release-notes.adoc (hunk 10 above)
Comment (maxwelldb) [reply release_notes/ocp-4-12-release-notes.adoc:2344]:
```suggestion
As a workaround, restart `openshift-cloud-controller-manager` pods by running the following command:
```

File: release_notes/ocp-4-12-release-notes.adoc (hunk 9 above)
Comment (maxwelldb) [reply release_notes/ocp-4-12-release-notes.adoc:2332]: I can find both styles throughout our docs, and nothing really definitive in the guidelines. Going to leave this as it is with v. little confidence. 

No. I'm going to change it. Again, though, very little confidence. Very much a gut thing. :shrug: 

```suggestion
* Bulk load-balancer member updates on {rh-openstack} can return a 500 code in response to `PUT` requests. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2100135[*BZ#2100135*])
```

File: release_notes/ocp-4-12-release-notes.adoc (hunk 11 above)
Comment (maxwelldb) [reply release_notes/ocp-4-12-release-notes.adoc:2353]: Arguably. Gonna go with it. 

```suggestion
* There is a known issue when `cloud-provider-openstack` tries to create health monitors on OVN load balancers by using the API to create fully populated load balancers. These health monitors become stuck in a `PENDING_CREATE` status. After their deletion, associated load balancers are are stuck in a `PENDING_UPDATE` status. There is no workaround. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2143732[*BZ#2143732*])
```

File: release_notes/ocp-4-12-release-notes.adoc [hunk 12]
```diff
@@ -654,6 +661,18 @@ As an administrator, you can now install the Network Observability Operator to o
 
 For more information, see xref:../networking/network_observability/network-observability-overview.html#network-observability-overview[Network Observability].
 
+[id="ocp-4-12-nw-shiftstack-ipv6-secondary"]
+==== IPv6 for secondary network interfaces on {rh-openstack}
+
+IPv6 for secondary network intefaces is now supported in clusters that run on {rh-openstack}.
```
Comment (eurijon) [inline release_notes/ocp-4-12-release-notes.adoc:667]:
```suggestion
IPv6 for secondary network interfaces is now supported in clusters that run on {rh-openstack}. 
```

File: release_notes/ocp-4-12-release-notes.adoc [hunk 13]
```diff
@@ -654,6 +661,18 @@ As an administrator, you can now install the Network Observability Operator to o
...
+==== IPv6 for secondary network interfaces on {rh-openstack}
+
+IPv6 for secondary network intefaces is now supported in clusters that run on {rh-openstack}. 
+
+// TODO: For more information,see ${DOC_WHEN_MERGED} 
+
+[id="ocp-4-12-nw-shiftstack-udp-load-balancing"]
+==== UDP support for load balancers on {rh-openstack}
+
+Resulting from the switch to an external {rh-openstack} cloud provider, UDP is now supported for load balancers for clusters that run on that platform.
```
Comment (eurijon) [inline release_notes/ocp-4-12-release-notes.adoc:674]:
```suggestion
Resulting from the switch to an external OpenStack cloud provider, UDP is now supported for load balancer type services for clusters that run on that platform.
```

File: release_notes/ocp-4-12-release-notes.adoc [hunk 14]
```diff
@@ -2288,6 +2327,31 @@ $ ./openshift-install wait-for install-complete
 
 * When using the Technology Preview OCI feature with the oc-mirror CLI plug-in, the mirrored catalog embeds all of the Operator bundles, instead of filtering only on those specified in the image set configuration file. (link:https://issues.redhat.com/browse/OCPBUGS-5085[*OCPBUGS-5085*])
 
+* Fully populated load balancers on {rh-openstack} that are created with the OVN driver can contain pools that are stuck in a pending creation status. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2042976[*BZ#2042976*])
```
Comment (eurijon) [inline release_notes/ocp-4-12-release-notes.adoc:2330]: @MaysaMacedo @dulek this issue has been solved in the latest z-streams, right? so this is here in case a customer is using a z-stream without the fix?

File: release_notes/ocp-4-12-release-notes.adoc [hunk 15]
```diff
@@ -2288,6 +2327,31 @@ $ ./openshift-install wait-for install-complete
...
+
+* Bulk load-balancer member updates on {rh-openstack} can return a 500 code in response to `PUT` requests. This issue can cause problems for clusters that are deployed on {rh-openstack}. To resolve the issue, update your {rh-openstack} packages. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2100135[*BZ#2100135*])
```
Comment (eurijon) [inline release_notes/ocp-4-12-release-notes.adoc:2332]: @MaysaMacedo @dulek same here

File: release_notes/ocp-4-12-release-notes.adoc (hunk 14 above)
Comment (MaysaMacedo) [reply release_notes/ocp-4-12-release-notes.adoc:2330]: @eurijon yes

File: release_notes/ocp-4-12-release-notes.adoc (hunk 15 above)
Comment (MaysaMacedo) [reply release_notes/ocp-4-12-release-notes.adoc:2332]: yes

File: release_notes/ocp-4-12-release-notes.adoc (hunk 13 above)
Comment (maxwelldb) [reply release_notes/ocp-4-12-release-notes.adoc:674]: @eurijon I find this phrasing a little hard to parse. As an alternative, would you prefer "services that have the load balancer type" or "`LoadBalancer` services" here?

Comment (eurijon) [reply release_notes/ocp-4-12-release-notes.adoc:674]: +1 for "LoadBalancer services" :)

Comment (maxwelldb) [reply release_notes/ocp-4-12-release-notes.adoc:674]:
```suggestion
Resulting from the switch to an external OpenStack cloud provider, UDP is now supported for `LoadBalancer` services for clusters that run on that platform.
```

**Comment by eurijon** [comment]
lgtm

**Review by stephenfin — COMMENTED** [review]
/lgtm

**Comment by maxwelldb** [comment]
Last call for change requests: Whenever lunchtime is for you, Jan. 13 :sandwich:

**Comment by MaysaMacedo** [comment]
It LGTM.
Just had one tiny question to @EmilienM

**Review by stephenfin — COMMENTED** [review]
/lgtm

**Comment by maxwelldb** [comment]
> It LGTM. Just had one tiny question to @EmilienM

@MaysaMacedo Sure. Might be missing, but I don't see it here. Will it be answered elsewhere?

File: release_notes/ocp-4-12-release-notes.adoc [hunk 16]
```diff
@@ -721,6 +728,18 @@ As an administrator, you can now install the Network Observability Operator to o
 
 For more information, see xref:../networking/network_observability/network-observability-overview.html#network-observability-overview[Network Observability].
 
+[id="ocp-4-12-nw-shiftstack-ipv6-secondary"]
+==== IPv6 for secondary network interfaces on {rh-openstack}
+
+IPv6 for secondary network interfaces is now supported in clusters that run on {rh-openstack}. 
+
+// TODO: For more information,see ${DOC_WHEN_MERGED}
```
Comment (maxwelldb) [inline release_notes/ocp-4-12-release-notes.adoc:736]:
```suggestion
For more information,see xref:../post_installation_configuration/network-configuration.adoc#nw-osp-pod-connections-ipv6_post-install-network-configuration[Enabling IPv6 connectivity to pods on {rh-openstack}].
```
